pkg/operator/ceph/object/notification/provisioner.go

```diff
@@ -79,61 +74,25 @@ func (p *Provisioner) getCephUser(username string, objStore *cephv1.CephObjectSt
 	return
 }
 
-func (p *Provisioner) createSession(owner string, objectStoreName types.NamespacedName) (*awssession.Session, error) {
-	objStore, err := p.Context.RookClientset.CephV1().CephObjectStores(objectStoreName.Namespace).Get(p.opManagerContext, objectStoreName.Name, metav1.GetOptions{})
+func newS3Agent(p Provisioner) (*object.S3Agent, error) {
+	objStore, err := p.context.RookClientset.CephV1().CephObjectStores(p.objectStoreName.Namespace).Get(p.opManagerContext, p.objectStoreName.Name, metav1.GetOptions{})
 	if err != nil {
-		return nil, errors.Wrapf(err, "failed to get CephObjectStore %v", objectStoreName)
+		return nil, errors.Wrapf(err, "failed to get CephObjectStore %v", p.objectStoreName)
 	}
 
-	objContext, err := object.NewMultisiteContext(p.Context, p.ClusterInfo, objStore)
+	objContext, err := object.NewMultisiteContext(p.context, p.clusterInfo, objStore)
 	if err != nil {
-		return nil, errors.Wrapf(err, "failed to get object context for CephObjectStore %v", objectStoreName)
+		return nil, errors.Wrapf(err, "failed to get object context for CephObjectStore %v", p.objectStoreName)
 	}
+	// CephClusterSpec is needed for GetAdminOPSUserCredentials()
+	objContext.CephClusterSpec = *p.clusterSpec
 
-	accessKey, secretKey, err := p.getCephUser(owner, objStore, objContext)
+	accessKey, secretKey, err := getUserCredentials(p.opManagerContext, p.owner, objStore, objContext)
 	if err != nil {
-		return nil, errors.Wrapf(err, "failed to get owner credentials for %q", owner)
-	}
-
-	// pass log level to AWS session
-	logLevel := aws.LogOff
-	if logger.LevelAt(capnslog.DEBUG) {
-		logLevel = aws.LogDebugWithHTTPBody
-	}
-
-	// pass TLS indication and certificates to AWS session
-	client := http.Client{
-		Timeout: object.HttpTimeOut,
-	}
-	tlsEnabled := objStore.Spec.IsTLSEnabled()
-	if tlsEnabled {
-		tlsCert := objContext.Context.KubeConfig.CertData
-		if len(tlsCert) > 0 {
-			client.Transport = object.BuildTransportTLS(tlsCert, false)
-		}
-	}
-
-	session, err := awssession.NewSession(
-		aws.NewConfig().
-			WithHTTPClient(&client).
-			WithRegion(objContext.ZoneGroup).
-			WithCredentials(credentials.NewStaticCredentials(accessKey, secretKey, "")).
-			WithEndpoint(objContext.Endpoint).
-			WithMaxRetries(3).
-			WithDisableSSL(!tlsEnabled).
-			WithS3ForcePathStyle(true).
-			WithLogLevel(logLevel),
-	)
-	if err != nil {
-		return nil, errors.Wrapf(err, "failed to create a new session for CephBucketNotification provisioning with %q", objectStoreName)
+		return nil, errors.Wrapf(err, "failed to get owner credentials for %q", p.owner)
 	}
 
-	logger.Debugf("session created. endpoint %q region %q secure %v",
-		*session.Config.Endpoint,
-		*session.Config.Region,
-		tlsEnabled,
-	)
-	return session, nil
+	return object.NewS3Agent(accessKey, secretKey, objContext.Endpoint, objContext.ZoneGroup, logger.LevelAt(capnslog.DEBUG), objContext.Context.KubeConfig.CertData)
```

Comment:
Why it is kept always as debug level for log? 
Will it flood the logs with s3 request details

Reply:
this is how it is used in: https://github.com/rook/rook/blob/master/pkg/operator/ceph/object/bucket/provisioner.go